hub/core/chunk_engine.py

```diff
@@ -279,26 +304,34 @@ def commit_diff_exists(self) -> bool:
 
     @property
     def chunk_id_encoder_exists(self) -> bool:
+        commit_id = self.commit_id
+        if (
+            self._chunk_id_encoder is not None
+            and self._chunk_id_encoder_commit_id == commit_id
+        ):
+            return True
         try:
-            commit_id = self.version_state["commit_id"]
             key = get_chunk_id_encoder_key(self.key, commit_id)
             self.meta_cache[key]
             return True
         except KeyError:
             return False
 
     def _is_tiled_sample(self, global_sample_index):
-        return self.tile_encoder_exists and global_sample_index in self.tile_encoder
+        return global_sample_index in self.tile_encoder
 
     @property
     def tile_encoder(self) -> TileEncoder:
         """Gets the tile encoder from cache, if one is not found it creates a blank encoder."""
-        commit_id = self.version_state["commit_id"]
+        commit_id = self.commit_id
         if self._tile_encoder is None or self._tile_encoder_commit_id != commit_id:
             key = get_tensor_tile_encoder_key(self.key, commit_id)
             if not self.tile_encoder_exists:
                 enc = TileEncoder()
-                self.meta_cache[key] = enc
+                try:
+                    self.meta_cache[key] = enc
+                except ReadOnlyModeError:
+                    pass
```

Comment:
If tile encoder does not exist and the storage is in read only mode, I am not sure we should be setting / returning an empty encoder that is not persistent. Should probably return None (and the calling method should check for None).

Reply:
checking for None won't work as that is the default value. To prevent multiple reads from storage we're checking self._tile_encoder to not be None. Persistence shouldn't be an issue as this only happens for read only.
If the user switches mode and tries to write, we will synchronize when the user does his first write in add_cachables_to_cache_dirty_keys